cmd/notary/tuf.go

```diff
@@ -834,3 +839,30 @@ func (a *authRoundTripper) RoundTrip(req *http.Request) (*http.Response, error)
 	// Return the last response
 	return resp, nil
 }
+
+func publish(cmd *cobra.Command, doPublish bool, gun string, config *viper.Viper, passRetriever notary.PassRetriever) error {
```

Comment:
Non-blocking - it's unfortunate that we can't inject a `RoundTripper` into an existing `NotaryRepository`.  

I'm wondering whether we can just have `publish` take a `NotaryRepository`, and refactor this function into a general `NotaryRepository` builder that creates a `NotaryRepository` with either a `nil` transport or a real transport, depending on whether it needs to be online. 

Also, as a future refactor we can do for `NotaryRepository`, I'm wondering if we should have the online functions `Publish`, `List`, etc. take a `RoundTripper` as an argument as opposed to having one stored on the `NotaryRepository`.